deltachat-ios/AppDelegate.swift

```diff
@@ -323,6 +329,28 @@ class AppDelegate: UIResponder, UIApplicationDelegate, UNUserNotificationCenterD
             }
         }
     }
+    
+    private var shouldShutdownEventLoop: Bool = false
+    private var eventShutdownSemaphore = DispatchSemaphore(value: 1)
```

Comment:
are you sure, the initial value is correct here?

the [condition for `wait()`](https://developer.apple.com/documentation/dispatch/1453087-dispatch_semaphore_wait) seems to be always always fulfilled, whether [`signal()`](https://developer.apple.com/documentation/dispatch/1452919-dispatch_semaphore_signal) is called or not

(i did not test or investigate deeper, it is just the rule-of-thumb that a semaphore with intial=1 is used for critical-sections and not for inter-thread-communicatio)

Reply:
I just copied this from somewhere and it did what I wanted, I did not dive far into the documentation.
https://developer.apple.com/documentation/dispatch/dispatchsemaphore

Reply:
>  it did what I wanted, I did not dive far into the documentation

well, did it really ever wait as you wanted? without the change, it may be that accounts are closed while still being in the event loop - maybe it just did never happen (can it?) with the change to value=0, you'll also get a warning that a user-qos-threads should not wait for a background-qos-thread (what, however, is okay in case of shutdown), indicating that there is really some wait taking effect

Reply:
> with the change to value=0, you'll also get a warning that a user-qos-threads should not wait for a background-qos-thread (what, however, is okay in case of shutdown),

I always get this warning even with value 1.
Isn't this just the start value? but maybe 0 is better?

Reply:
0 seems to be more correct from what I read online, maybe just change it to 0 then.

Reply:
yeah let's change it to 0